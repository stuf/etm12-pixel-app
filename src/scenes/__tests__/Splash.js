import * as React from 'karet';
import * as K from 'kefir';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SplashScene from 'scenes/Splash';

it('renders correctly', () => {
  const wrapper = shallow(<SplashScene />);

  expect(toJson(wrapper)).toMatchSnapshot();
});

it("doesn't trigger redirect if no target is specified", done => {
  const history = {
    replace: jest.fn(),
  };

  const wrapper = mount(<SplashScene history={history} redirectDelay={0} />);

  K.later(50, null).onValue(() => {
    expect(history.replace).not.toHaveBeenCalled();
    done();
  });
});

it('triggers configured redirect', done => {
  const history = {
    replace: jest.fn(),
  };

  const wrapper = mount(
    <SplashScene history={history} redirectTo="/foo" redirectDelay={0} />,
  );

  K.later(50, null).onValue(() => {
    expect(history.replace).toHaveBeenCalled();
    done();
  });
});
