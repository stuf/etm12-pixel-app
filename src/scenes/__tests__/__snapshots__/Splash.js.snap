// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="scene-root splash-root"
  data-scene-name="splash"
>
  <FromClass
    args={
      Array [
        Symbol(react.fragment),
        Object {},
        Combine {
          "_activating": false,
          "_active": false,
          "_alive": true,
          "_currentEvent": null,
          "_dispatcher": Dispatcher {
            "_inLoop": 0,
            "_items": Array [],
            "_removedItems": null,
            "_spies": Array [],
          },
          "_f": [Function],
          "_h": null,
          "_logHandlers": null,
          "_n": 1,
          "_spyHandlers": null,
          "_x": Array [
            AnonymousObservable {
              "_$handleAny": [Function],
              "_activating": false,
              "_active": false,
              "_alive": true,
              "_currentEvent": null,
              "_dispatcher": Dispatcher {
                "_inLoop": 0,
                "_items": Array [],
                "_removedItems": null,
                "_spies": Array [],
              },
              "_getInitialCurrent": [Function],
              "_logHandlers": null,
              "_name": "constant.toProperty",
              "_source": P {
                "_currentEvent": Object {
                  "current": true,
                  "type": "value",
                  "value": undefined,
                },
              },
              "_spyHandlers": null,
            },
          ],
        },
      ]
    }
  />
  <div
    className="splash-logo"
  >
    <img
      alt="pixel"
      src="logo.svg"
    />
    <h1>
      pixel
    </h1>
  </div>
</div>
`;
